src/mame/apple/apple2video.cpp

```diff
@@ -713,38 +606,34 @@ void a2_video_device::text_update(screen_device &screen, bitmap_ind16 &bitmap, c
 
 	// printf("TXT: row %d startcol %d stopcol %d left %d right %d\n", beginrow, startcol, stopcol, cliprect.left(), cliprect.right());
 
-	int fg = 0;
-	int bg = 0;
-	switch (m_sysconfig & 0x03)
+	int fg, bg;
+	if (model == Model::IIGS)
 	{
-	case 0:
-	case 4:
-		fg = WHITE;
-		break;
-	case 1:
-		fg = WHITE;
-		break;
-	case 2:
-		fg = GREEN;
-		break;
-	case 3:
-		fg = ORANGE;
-		break;
+		fg = m_GSfg;
+		bg = m_GSbg;
+	}
+	else
+	{
+		switch (m_sysconfig & 0x03)
+		{
+			case 2: fg = GREEN; break;
+			case 3: fg = ORANGE; break;
+			default: fg = WHITE; break;
+		}
+		bg = 0;
 	}
 
 	for (int row = startrow; row < stoprow; row += 8)
 	{
-		if ((iie) && (m_80col))
+		if ((model == Model::IIe || model == Model::IIGS) && m_80col)
 		{
 			for (int col = startcol; col < stopcol; col++)
 			{
 				/* calculate address */
 				uint32_t const address = start_address + ((((row / 8) & 0x07) << 7) | (((row / 8) & 0x18) * 5 + col));
-
-				plot_text_character<true, invert, flip>(bitmap, col * 14, row, 1, aux_page[address & m_aux_mask],
-									fg, bg);
-				plot_text_character<true, invert, flip>(bitmap, col * 14 + 7, row, 1, m_ram_ptr[address],
-									fg, bg);
+				uint32_t aux_address = (model == Model::IIGS) ? address : address & m_aux_mask;
+				plot_text_character<model, invert, flip>(bitmap, col * 14, row, 1, aux_page[aux_address], fg, bg);
```

Comment:
Can `aux_address` be made `const`?  Also would be a bit clearer with parentheses around `(address & m_aux_mask)`.